### PR TITLE
Support for build analysers scheduling

### PR DESCRIPTION
**Pull request by harshad16**
- Support for build analysers scheduling
- Auto black changes 
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

### CODE REVIEW

File: thoth/common/openshift.py [hunk 1]
```diff
@@ -1136,6 +1142,220 @@ def get_dependency_monkey_template(self):
 
         return self._get_template("template=dependency-monkey")
 
+    def schedule_build_analyse(
+        self,
+        document_id: str,
+        output: str,
+        output_format: str = "json",
```
Comment (fridex) [inline thoth/common/openshift.py]: BTW I would hardcode this to OpenShift template, we will not produce anything else than JSON in the cluster.

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -1136,6 +1142,220 @@ def get_dependency_monkey_template(self):
...
+        job_id: str = None,
+    ) -> str:
+        """Schedule an build analyse run."""
+        if not self.backend_namespace:
```
[the remark this answers is not in the document]
Comment (harshad16) [reply thoth/common/openshift.py]: Thanks for the review and explanation. I will change the namespace to middletier :+1:

File: thoth/common/openshift.py [hunk 3]
```diff
@@ -1136,6 +1142,220 @@ def get_dependency_monkey_template(self):
...
+            raise ConfigurationError(
+                "Unable to schedule build analyse without backend namespace being set"
+            )
+
+        job_id = job_id or self._generate_id("build-analyse")
```
[the remark this answers is not in the document]
Comment (harshad16) [reply thoth/common/openshift.py]: It is name based upon persona - build-analyse, build-report, and build-dependencies. 
Each with a different use-case.

File: thoth/common/openshift.py (hunk 1 above)
Comment (harshad16) [reply thoth/common/openshift.py]: ACK, thanks will include the value of output_format to template and make the changes to files accordingly.

**Pull request merged**